Manifold/Type.swift

```diff
@@ -14,26 +18,187 @@ public enum Type: Hashable {
 	}
 
 
+	public static var Bool: Type {
+		return Type(.Bool)
+	}
+
+	public static var Unit: Type {
+		return Type(.Unit)
+	}
+
+
+	public enum Constructor: Hashable, Printable {
+		case Unit
+		case Bool
+		case Function(Type, Type)
```

Comment:
During the development of this PR, I had introduced a `BaseType` `enum` here which provided `Unit` and `Bool`. However, when I implemented typing errors for attempts to apply `true` as though it were a function, it occurred to me that the check for mutually exclusive types was actually a check for mutually exclusive type _constructors_.

This also provides a natural place to add sum/product types and other interesting type constructors.